### ISSUE TITLE
Can't build with libFLAME

### ISSUE THREAD

**Issue by s-sajid-ali** [numpy/numpy#14205]
`numpy` fails to link to any variant of libFLAME (release, master, static, shared) and complains of undefined references of BLAS functions from the libflame library. When I attempted to build `numpy+blis+libflame`, I ended up building `numpy+blis~libflame`

### Reproducing code example:
Using spack to build numpy as per this [PR](https://github.com/spack/spack/pull/12170)

Specific commands executed during build&install are as follows 
`python3.7' '-s' 'setup.py' '--no-user-cfg' 'build'`

`python3.7' '-s' 'setup.py' '--no-user-cfg' 'install' '--prefix=/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/py-numpy-1.17.0-psnrtylx4lzvesb6zovmdim7mlt2ubh6' '--single-version-externally-managed' '--root=/`

[site.cfg](https://pastebin.com/WT8qewrQ) generated during the build process.
[build.log](http://pasted.co/7d0db170)
[build.env](http://pasted.co/250a78a2)

libFLAME was configured to build with lapack2flame and generate a static library ( [build.log](http://pasted.co/d14de1db) for the same).

### Numpy/Python version information:

```
>>> print(numpy.__version__, sys.version)
1.17.0 3.7.4 (default, Aug  5 2019, 20:17:04)
[GCC 9.1.0]
```

Link info : 
```
Python version                                                                                                                                        
                                                                                                                                                      
3.7.4 (default, Aug  5 2019, 20:17:04)                                                                                                                
[GCC 9.1.0]                                                                                                                                           
                                                                                                                                                      
OS info                                                                                                                                               
                                                                                                                                                      
Linux-3.10.0-693.el7.x86_64-x86_64-with-centos-7.5.1804-Core                                                                                          
                                                                                                                                                      
Numpy BLAS/Lapack linkage                                                                                                                             
                                                                                                                                                      
blis_info:                                                                                                                                            
    libraries = ['blis', 'blis']                                                                                                                      
    library_dirs = ['/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/blis-0.6.0-4nabwbykywmmdbrugpp46y74awliplpp/lib']            
    define_macros = [('HAVE_CBLAS', None)]                                                                                                            
    include_dirs = ['/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/blis-0.6.0-4nabwbykywmmdbrugpp46y74awliplpp/include']        
    language = c                                                                                                                                      
    runtime_library_dirs = ['/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/blis-0.6.0-4nabwbykywmmdbrugpp46y74awliplpp/lib']    
blas_opt_info:                                                                                                                                        
    libraries = ['blis', 'blis']                                                                                                                      
    library_dirs = ['/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/blis-0.6.0-4nabwbykywmmdbrugpp46y74awliplpp/lib']            
    define_macros = [('HAVE_CBLAS', None)]                                                                                                            
    include_dirs = ['/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/blis-0.6.0-4nabwbykywmmdbrugpp46y74awliplpp/include']        
    language = c                                                                                                                                      
    runtime_library_dirs = ['/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-9.1.0/blis-0.6.0-4nabwbykywmmdbrugpp46y74awliplpp/lib']    
flame_info:                                                                                                                                           
  NOT AVAILABLE                                                                                                                                       
lapack_opt_info:                                                                                                                                      
  NOT AVAILABLE                                                                                                                                       
```                                                           
                                                                                           
PS : I'm not sure if this error is due incorrect build process of numpy or with libflame itself. If this is due to an incorrectly configured libflame, feel free to close this issue.

**Comment by adamjstewart** [comment]
I encountered this same problem as well. With a `site.cfg` like:
```
[flame]
libraries = flame
library_dirs = /home/adamjs5/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/libflame-master-xmo2dhpugpfgb4zetev2lcccvdgybjog/lib
rpath = /home/adamjs5/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/libflame-master-xmo2dhpugpfgb4zetev2lcccvdgybjog/lib
include_dirs = /home/adamjs5/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/libflame-master-xmo2dhpugpfgb4zetev2lcccvdgybjog/include
```
the numpy build did not pick up the flame library. Note that I explicitly set:
```bash
export NPY_BLAS_ORDER=''
export NPY_LAPACK_ORDER=flame
```

**Comment by rgommers** [comment]
Probably related to BLIS build issues (gh-14180), so cross-linking.

**Comment by rgommers** [comment]
@adamjstewart did this get solved as well now that you figured out the issue with BLIS?

**Comment by adamjstewart** [comment]
Let me try again.

**Comment by adamjstewart** [comment]
@rgommers I tried several compilers and versions of libflame on multiple OSes, but couldn't get libflame to install successfully. See https://github.com/flame/libflame/issues/23 and https://github.com/flame/libflame/issues/24. Given how broken libflame is, I wouldn't worry too much about whether or not there is a problem on the numpy side of things until we can get these issues resolved.

**Comment by rgommers** [comment]
thanks! 

> Given how broken libflame is, I wouldn't worry too much about whether or not there is a problem on the numpy side of things until we can get these issues resolved.

yes I agree, removing the milestone

**Comment by adamjstewart** [comment]
https://github.com/flame/libflame/issues/24 is now fixed. With the changes I made in https://github.com/spack/spack/pull/12919, I can now install NumPy with BLIS and libflame on CentOS 7 with GCC 7.2.0 and Python 3.7.4. All unit tests pass.

The macOS build of libflame is still broken, but that's been reported to libflame. As far as I'm concerned, this issue can probably be closed. Thanks for your help @rgommers!
